Python/marshal.c

```diff
@@ -875,17 +925,60 @@ r_long64(RFILE *p)
+static int                                                              \
+_w_digits##bs(uint##bs##_t *digits, Py_ssize_t size,                    \
+              Py_ssize_t marshal_ratio,                                 \
+              int shorts_in_top_digit, RFILE *p)                        \
+{                                                                       \
+    int md;                                                             \
+    uint##bs##_t d;                                                     \
+                                                                        \
+    for (Py_ssize_t i = 0; i < size - 1; i++) {                         \
+        d = 0;                                                          \
+        for (Py_ssize_t j = 0; j < marshal_ratio; j++) {                \
+            md = r_short(p);                                            \
+            if (md < 0 || md > PyLong_MARSHAL_BASE) {                   \
+                goto bad_digit;                                         \
+            }                                                           \
+            d += (uint##bs##_t)md << j*PyLong_MARSHAL_SHIFT;            \
+        }                                                               \
+        digits[i] = d;                                                  \
+    }                                                                   \
+                                                                        \
+    d = 0;                                                              \
+    for (Py_ssize_t j = 0; j < shorts_in_top_digit; j++) {              \
+        md = r_short(p);                                                \
+        if (md < 0 || md > PyLong_MARSHAL_BASE) {                       \
+            goto bad_digit;                                             \
+        }                                                               \
+        /* topmost marshal digit should be nonzero */                   \
+        if (md == 0 && j == shorts_in_top_digit - 1) {                  \
+            PyErr_SetString(PyExc_ValueError,                           \
+                "bad marshal data (unnormalized long data)");           \
+            return -1;                                                  \
+        }                                                               \
+        d += (uint##bs##_t)md << j*PyLong_MARSHAL_SHIFT;                \
+    }                                                                   \
+    assert(!PyErr_Occurred());                                          \
+    /* top digit should be nonzero, else the resulting PyLong won't be  \
+       normalized */                                                    \
+    digits[size - 1] = d;                                               \
+    return 0;                                                           \
+bad_digit:                                                              \
```

Comment:
Add a newline for readability:

```suggestion
    return 0;                                                           \
                                                                        \
bad_digit:                                                              \
```